src/treeview_commands.ml

```diff
@@ -46,16 +46,21 @@ let getChildren ~extension_path ~element =
   | None -> `Value (Some (items ~extension_path))
   | Some _ -> `Value (Some [])
 
-let register extension =
-  let extension_path = Vscode.ExtensionContext.extensionPath extension in
-  let treeDataProvider =
-    Vscode.TreeDataProvider.create
-      ~getTreeItem:(getTreeItem ~extension_path)
-      ~getChildren:(getChildren ~extension_path)
-      ()
+let register extension (_ : Extension_instance.t) =
+  let (_ : unit Promise.t) =
+    Promise.make @@ fun ~resolve ~reject:_ ->
```

Comment:
Unless my understanding of JS is woefully out of date, this is no different than using `Promise.return`. No additionally concurrency is gained from writing the code this way.